lib/topics/index.ts

```diff
@@ -0,0 +1,177 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import { HandlerFn, IRequestOptions, ISubscription, Mediator } from '../mediator';
+
+export interface IErrorWithDataId extends Error {
+  /** Identifier to use as a suffix on the 'error:' topic */
+  id?: string;
+}
+
+interface IHasId {
+  id?: string;
+}
+
+export default class Topics {
+  public mediator: Mediator;
+  public prefix: string;
+  public entity: string;
+  private subscriptions: {
+    [key: string]: ISubscription;
+  };
+
+  constructor (mediator: Mediator) {
+    this.mediator = mediator;
+    this.subscriptions = {};
+  }
+
+  /**
+   * Sets the prefix configuration for this instance, which will be part of the name
+   * of the handled topics.
+   * @param  {String} prefix
+   * @return {Topics}        returns self for chaining
+   */
+  public withPrefix(prefix: string) {
```

Comment:
My OCD-ness also tingles when i add inconsistencies like this :D 

I actually think we should go without them and leverage the type inference more. (ts correctly infers the type on both instances)